### DISCUSSION TITLE
Datadir location and /mnt/ncdata on the host

### DISCUSSION THREAD

**Discussion by loeffelpan** [nextcloud/all-in-one#1644] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Setup AIO from scratch with any path for DATADIR outside `/mnt/ncdata` (including default docker volume path).
I used `/opt/ncdata` with `NEXTCLOUD_DATADIR` env-var.
2. `/mnt/ncdata` **on the host** is created anyway.
3. Delete `/mnt/ncdata` **on the host**.

### Expected behavior 
Nothing should happen, as `/mnt/ncdata` **on the host** is unused.

### Actual behavior 
Nextcloud shows "Dein Datenverzeichnis ist ungültig. Stelle sicher, dass eine Datei ".ocdata" im Wurzelverzeichnis des data-Verzeichnisses existiert."

### Host OS 
Ubuntu Server 20.04.5 LTS

#### Nextcloud AIO version 
Nextcloud AIO v4.0.1

#### Current channel 
latest

#### Additional information #####
`/mnt/ncdata` is the location of the datadir **inside the nextcloud container**. Maybe that's a lead to this problem.

**Comment by szaimen** [comment]
Hi, can you post the output of `sudo docker inspect nextcloud-aio-nextcloud | grep "/mnt/ncdata"`?

**Comment by loeffelpan** [comment]
I already did that, but there are nothing suspicious, I think.
`/mnt` is my mount-path for external storage.

```
nightfury@ToothLess:~$ sudo docker inspect nextcloud-aio-nextcloud | grep ncdata
                "/opt/ncdata:/mnt/ncdata:rw",
                "Source": "/opt/ncdata",
                "Destination": "/mnt/ncdata",
                "NEXTCLOUD_DATA_DIR=/mnt/ncdata",
                "/mnt/ncdata": {},
```

`ncdata` appears in this JSON-parts:

`HostConfig`
```
"Binds": [
                "nextcloud_aio_nextcloud:/var/www/html:rw",
                "/opt/ncdata:/mnt/ncdata:rw",
                "/mnt:/mnt:rw"
            ]
```
```
"Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud",
                "Source": "/var/lib/docker/volumes/nextcloud_aio_nextcloud/_data",
                "Destination": "/var/www/html",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/opt/ncdata",
                "Destination": "/mnt/ncdata",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            },
            {
                "Type": "bind",
                "Source": "/mnt",
                "Destination": "/mnt",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ]
```

`Config` in `HostConfig`
```
"Env": [
                "NEXTCLOUD_DATA_DIR=/mnt/ncdata",
            ]
```
```
"Volumes": {
                "/mnt/ncdata": {},
                "/var/www/html": {}
            },
```

**Comment by loeffelpan** [comment, marked as answer]
I think binding `/mnt` for external storage is the problem. This let's `/mnt/ncdata` break out of the container. Maybe I have to mount my external disk at `/mnt/Daten` and bind this via `NEXTCLOUD_MOUNT` into container to avoid this conflict.

**Comment by szaimen** [comment]
> I think binding `/mnt` for external storage is the problem. This let's `/mnt/ncdata` break out of the container. Maybe I have to mount my external disk at `/mnt/Daten` and bind this via `NEXTCLOUD_MOUNT` into container to avoid this conflict.

Ah this makes sense. Moving to discussions since I dont see anything we can improve on our side.
